torch_geometric/nn/models/explainer.py

```diff
@@ -114,8 +114,127 @@ def forward(self, mask, *args):
         return x
 
 
+# TODO(jinu) can captum work with list of tensors
+# probably need to convert to Tuple
+def hetero_data_to_captum_data(data: HeteroData,
+                               mask_type: str) -> List[Tensor]:
+    """Takes data.x_dict and data.edge_index_dict and converts
+    it ot a list of tensors for use by captum."""
+    node_types = data.metadata[0]
+    edge_types = data.metadata[1]
+    if mask_type == "node":
+        explain_types = node_types
+    elif mask_type == "edge":
+        explain_types = edge_types
+    elif mask_type == "node_and_edge":
+        explain_types = node_types + edge_types
+    else:
+        raise ValueError(f"Invalid mask_type (got {mask_type})")
+    out_tensors = []
+    for key in explain_types:
+        out_tensors.append(data[key])
+    return out_tensors
+
+
+def captum_output_to_dicts(
+    captum_attrs: Tuple[Tensor], mask_type, metadata: Metadata
+) -> Tuple[Optional[Dict[NodeType, Tensor]], Optional[Dict[EdgeType, Tensor]]]:
+    """Convert the output of Captum.attribute to `x_attr_dict` and
+    `edge_attr_dict`"""
+    node_types = metadata[0]
+    edge_types = metadata[1]
+    x_attr_dict, edge_attr_dict = None, None
+    if mask_type == "node":
+        assert len(node_types) == len(captum_attrs)
+        x_attr_dict = dict(zip(node_types, captum_attrs))
+    elif mask_type == "edge":
+        assert len(edge_types) == len(captum_attrs)
+        edge_attr_dict = dict(zip(edge_types, captum_attrs))
+    elif mask_type == "node_and_edge":
+        assert len(edge_types) + len(node_types) == len(captum_attrs)
+        x_attr_dict = dict(zip(node_types, captum_attrs[0:len(node_types)]))
+        edge_attr_dict = dict(zip(edge_types, captum_attrs[len(node_types):]))
+    return x_attr_dict, edge_attr_dict
+
+
+# TODO(jinu) Is there any point of inheriting from `CaptumModel`
+class CaptumHeteroModel(CaptumModel):
+    def __init__(self, model: torch.nn.Module, mask_type: str, output_id: int,
+                 metadata: Metadata):
+        super().__init__(model, mask_type, output_id)
+        self.edge_types = metadata[0]
+        self.node_types = metadata[1]
+
+    def _captum_data_to_hetero_data(
+        self, *args
+    ) -> Tuple[Dict[NodeType, Tensor], Dict[EdgeType, Tensor], Optional[Dict[
+            EdgeType, Tensor]]]:
+        """Converts tuple of tensors to `x_dict`, `edge_index_dict` and
+        `edge_mask_dict`."""
+        num_edge_types = len(self.edge_types)
+        num_node_types = len(self.node_types)
+
+        if self.mask_type == 'node':
+            node_tensors = args[0:num_node_types]
+            edge_tensors = args[num_node_types:num_node_types + num_edge_types]
```

Comment:
This should be `args[num_node_types:num_node_types + 1]` as you would forward the edges in a normal dict format over the additional args and not over the input.

Reply:
Good point. updated the PR.